jupyter-extension/jupyterlab_pachyderm/pfs_manager.py

```diff
@@ -198,6 +200,9 @@ def __init__(self, client: Client, **kwargs):
         self._client = client
         super().__init__(**kwargs)
 
+    def mount_branch(self, branch: pfs.Branch):
+        self.mounted_branch = branch
```

Comment:
I think it makes sense for the `branch_exist` call to be made here so that this manager manages the validity of the mounted branch entirely.